library/std/src/sys/unix/fs.rs

```diff
@@ -54,6 +54,8 @@ use libc::c_char;
 use libc::dirfd;
 #[cfg(any(target_os = "linux", target_os = "emscripten"))]
 use libc::fstatat64;
+#[cfg(all(miri, any(target_os = "linux")))]
+use libc::open64;
```

Comment:
I'm a bit surprised that this import is only present for Miri -- that seems strange?

Reply:
There are two code-paths, one that uses openat and one that uses open. Miri only supports the latter, so the entire machinery around -at syscalls is configured-out under miri and the alternative is configured-in.

Reply:
Ah, and there are some other import for `open64` below for other targets. The entire set of `cfg`s is just tricky to have an overview of. Makes sense.